api-report/sequence.api.md

```diff
@@ -340,12 +358,16 @@ export interface ISerializableInterval extends IInterval {
 
 // @internal
 export interface ISerializedInterval {
-    canBeExclusive?: boolean;
+    canSlideToEndpoint?: boolean;
```

Comment:
We should avoid changing the persisted format as much as possible. I think the amount of state we need to convey in the long-term is actually just the interval's stickiness (or equivalently, the `startSide / endSide` values you have). Since we already committed to that, we should just keep it as the persisted format.

I need to refresh myself on the desire for `canBeExclusive` as part of this PR, but there should be some alternatives.

Reply:
The goal of `canSlideToEndpoint` is to make the new model of stickiness opt-in, such that the behavior of existing intervals doesn't change. This means we have to persist this value to determine whether it's a legacy interval or one with endpoints using the new sliding behavior. We could perhaps consolidate the serialized format to only the values of stickiness and isLegacy. The start/end sides are currently included in the serialized format for a similar reason.

Reply:
Makes sense. I'll push up a change adding a feature flag to merge-tree that opts-in to the sliding to endpoints.

I think we still need to keep track of sides and stickiness, however. Also for legacy reasons, it is possible to have an end-sticky interval that has both sides inclusive.